src/core/manager.c

```diff
@@ -1973,6 +1974,11 @@ int manager_startup(Manager *m, FILE *serialization, FDSet *fds, const char *roo
                                 return log_error_errno(r, "Deserialization failed: %m");
                 }
 
+                /* If we are in a new soft-reboot iteration bump the counter now before starting units, so
+                 * that they can reliably read it. We get the previous objective from serialized state. */
+                if (m->previous_objective == MANAGER_SOFT_REBOOT)
+                        m->soft_reboots_count++;
+
```

Comment:
I think it would be really good if we output the previous objective here as well as the soft reboot counter.

i.e.:

```c
if (m->previous_objective >= 0)
        log_info("Launching as effect of a '%s' operation.", manager_objective_to-string(m->previous_objective));
if (m->soft_reboots_count > 0)
        log_info("Soft reboot counter is now at %u.", m->soft_reboots_count);
```

I think this would make debugging things a lot easier.

Reply:
Not sure, especially at info level - but logs can be added at any time, let's get the APIs sorted first so we can do RC1 please

Reply:
(I like this, but one note: we should not log at > debug if the objective is reload, since we already log about that in `invoke_main_loop`)

> Not sure, especially at info level - but logs can be added at any time, let's get the APIs sorted first so we can do RC1 please

I'll send a follow-up.